src/app/modules/timesheet/components/invoice-form/invoice-form.component.ts

```diff
@@ -14,7 +18,91 @@ export class InvoiceFormComponent implements OnInit {
   @Input() invoice: Invoice;
   @Output() changed: EventEmitter<boolean> = new EventEmitter();
 
+  constructor ( private _route: ActivatedRoute,
+                private _missionService: MissionService,
+                private _timesheetService: TimesheetService) {}
+
   ngOnInit() {
+
+    this._missionService.readById(this._route.snapshot.params.missionId).then( response => {
+
+      if (response.data.consultantFreelance) {
+
+      this.invoice.provider.name = response.data.consultantCompany.name;
```

Comment:
You copy every same attribute in another place, it's not easier to do :
```this.invoice.provider = response.data.consultantCompany```
And you can delete the two - three excess fields with a `delete`
If the type is correct you will be protected and will not be able to do mistake

Or you can perhaps use the lodash _pick 
https://lodash.com/docs/4.17.11#pick